src/module/Entity/config/types.config.php

```diff
@@ -164,23 +182,158 @@
                             ]
                         ]
                     ],
-                    'license'    => []
+                    'license' => []
                 ]
             ],
-            'text-solution'         => [
-                'title'       => 'id',
+            'input-string-normalized-match-challenge' => [
+                'title' => 'id',
                 'description' => 'content',
-                'components'  => [
+                'components' => [
                     'repository' => [
-                        'form'   => __NAMESPACE__ . '\Form\TextSolutionForm',
+                        'form' => __NAMESPACE__ . '\Form\InputChallengeForm',
+                        'fields' => [
+                            'solution',
+                            'feedback',
+                            'changes',
+                        ]
+                    ],
+                    'link' => [
+                        'children' => [
+                            'input-string-normalized-match-challenge' => [
+                                'multiple' => true
+                            ],
+                            'input-number-exact-match-challenge' => [
+                                'multiple' => true
+                            ],
+                            'input-expression-equal-match-challenge' => [
+                                'multiple' => true
+                            ]
+                        ],
+                        'parents' => [
+                            'text-exercise' => [
+                                'multiple' => false
+                            ],
+                            'grouped-text-exercise' => [
+                                'multiple' => false
+                            ],
+                            'input-string-normalized-match-challenge' => [
```

Comment:
Well, new `input-*-match-challenge` types occur in a couple of places... Not sure how to fix this... Tried some stuff with `array_map` but failed at creating this associative array thingy. Any ideas for this? Ideally, we would have some array of the existing `input-*-match-challenge` types in one place so we can refer to it everywhere...